internal/service/templateversion.go

```diff
@@ -0,0 +1,127 @@
+package service
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/api/server"
+	"github.com/flightctl/flightctl/internal/store"
+	"github.com/go-openapi/swag"
+	"gorm.io/gorm"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func TemplateVersionFromReader(r io.Reader) (*api.TemplateVersion, error) {
+	var templateVersion api.TemplateVersion
+	decoder := json.NewDecoder(r)
+	decoder.DisallowUnknownFields()
+	err := decoder.Decode(&templateVersion)
+	return &templateVersion, err
+}
+
+// (POST /api/v1/templateVersions)
+func (h *ServiceHandler) CreateTemplateVersion(ctx context.Context, request server.CreateTemplateVersionRequestObject) (server.CreateTemplateVersionResponseObject, error) {
+	orgId := store.NullOrgId
+
+	result, err := h.store.TemplateVersion().Create(ctx, orgId, request.Body)
+
+	switch err {
+	case nil:
+		return server.CreateTemplateVersion201JSONResponse(*result), nil
+	case gorm.ErrRecordNotFound:
+		return server.CreateTemplateVersion400Response{}, nil
+	case gorm.ErrInvalidData:
+		return server.CreateTemplateVersion409Response{}, nil
+	case gorm.ErrDuplicatedKey:
+		return server.CreateTemplateVersion409Response{}, nil
+
+	default:
+		return nil, err
+	}
+}
+
+// (GET /api/v1/templateVersions)
+func (h *ServiceHandler) ListTemplateVersions(ctx context.Context, request server.ListTemplateVersionsRequestObject) (server.ListTemplateVersionsResponseObject, error) {
+	orgId := store.NullOrgId
+	labelSelector := ""
+	if request.Params.LabelSelector != nil {
+		labelSelector = *request.Params.LabelSelector
+	}
+
+	labelMap, err := labels.ConvertSelectorToLabelsMap(labelSelector)
+	if err != nil {
+		return nil, err
+	}
+
+	cont, err := store.ParseContinueString(request.Params.Continue)
+	if err != nil {
+		return server.ListTemplateVersions400Response{}, fmt.Errorf("failed to parse continue parameter: %s", err)
+	}
+
+	listParams := store.ListParams{
+		Labels:   labelMap,
+		Limit:    int(swag.Int32Value(request.Params.Limit)),
+		Continue: cont,
+		Owner:    request.Params.Owner,
+	}
+	if listParams.Limit == 0 {
+		listParams.Limit = store.MaxRecordsPerListRequest
+	}
+	if listParams.Limit > store.MaxRecordsPerListRequest {
+		return server.ListTemplateVersions400Response{}, fmt.Errorf("limit cannot exceed %d", store.MaxRecordsPerListRequest)
+	}
+
+	result, err := h.store.TemplateVersion().List(ctx, orgId, listParams)
+	switch err {
+	case nil:
+		return server.ListTemplateVersions200JSONResponse(*result), nil
+	default:
+		return nil, err
+	}
```

Comment:
```suggestion
	result, err := h.store.TemplateVersion().List(ctx, orgId, listParams)
        if err != nil{
           return err
        }
        return server.ListTemplateVersions200JSONResponse(*result), nil
}

Reply:
I definitely get what you're saying.  It's the convention in this project for all methods of all resources.  I assume the motivation was that there are some methods that deal with multiple error codes.  I opened MGMT-17112 to tackle this.